nacos-spring-context/src/main/java/com/alibaba/nacos/spring/util/parse/DefaultYamlConfigParse.java

```diff
@@ -49,189 +49,152 @@
-		buildFlattenedMap(result, source, null);
-		return result;
-	}
-
-	protected static void buildFlattenedMap(Map<String, Object> result,
-			Map<String, Object> source, String path) {
-		for (Map.Entry<String, Object> entry : source.entrySet()) {
-			String key = entry.getKey();
-			if (!StringUtils.isBlank(path)) {
-				if (key.startsWith("[")) {
-					key = path + key;
-				}
-				else {
-					key = path + '.' + key;
-				}
-			}
-			Object value = entry.getValue();
-			if (value instanceof String) {
-				result.put(key, value);
-			}
-			else if (value instanceof Map) {
-				// Need a compound key
-				@SuppressWarnings("unchecked")
-				Map<String, Object> map = (Map<String, Object>) value;
-				buildFlattenedMap(result, map, key);
-			}
-			else if (value instanceof Collection) {
-				// Need a compound key
-				@SuppressWarnings("unchecked")
-				Collection<Object> collection = (Collection<Object>) value;
-				int count = 0;
-				for (Object object : collection) {
-					buildFlattenedMap(result,
-							Collections.singletonMap("[" + (count++) + "]", object), key);
-				}
-			}
-			else {
-				result.put(key, (value != null ? value.toString() : ""));
-			}
-		}
-	}
-
-	@Override
-	public Map<String, Object> parse(String configText) {
-		final AtomicReference<Map<String, Object>> result = new AtomicReference<Map<String, Object>>();
-		process(new MatchCallback() {
-			@Override
-			public void process(Map<String, Object> map) {
-				result.set(map);
-			}
-		}, createYaml(), configText);
-		return result.get();
-	}
-
-	@Override
-	public String processType() {
-		return ConfigType.YAML.getType();
-	}
-
-	protected interface MatchCallback {
-
-		/**
-		 * Put Map to Properties.
-		 *
-		 * @param map {@link Map}
-		 */
-		void process(Map<String, Object> map);
-	}
-
-	protected static class MapAppenderConstructor extends Constructor {
-
-		MapAppenderConstructor() {
-			super();
-		}
-
-		@Override
-		protected Map<Object, Object> constructMapping(MappingNode node) {
-			try {
-				return super.constructMapping(node);
-			}
-			catch (IllegalStateException ex) {
-				throw new ParserException("while parsing MappingNode",
-						node.getStartMark(), ex.getMessage(), node.getEndMark());
-			}
-		}
-
-		@Override
-		protected Map<Object, Object> createDefaultMap(int initSize) {
-			final Map<Object, Object> delegate = super.createDefaultMap(initSize);
-			return new AbstractMap<Object, Object>() {
-				@Override
-				public Object put(Object key, Object value) {
-					if (delegate.containsKey(key)) {
-						throw new IllegalStateException("Duplicate key: " + key);
-					}
-					return delegate.put(key, value);
-				}
-
-				@Override
-				public Set<Entry<Object, Object>> entrySet() {
-					return delegate.entrySet();
-				}
-			};
-		}
-	}
-
+    
+    protected static final Logger LOGGER = LoggerFactory.getLogger(DefaultYamlConfigParse.class);
+    
+    private static final String YAML_ALLOW_COMPLEX_OBJECT = "yamlAllowComplexObject";
```

Comment:
I've noticed that similar configurations use camelCase naming conventions, such as ignoreResourceNotFound and ignoreUnresolvablePlaceholders, so I've adopted camelCase naming for consistency.